jack-sys/build.rs

```diff
@@ -11,4 +22,1991 @@ fn main() {
         // not installed, for example on Windows.
         pkg_config::find_library("jack").unwrap();
     }
+    write_src("src/functions.rs", FUNCTIONS);
+}
+
+fn write_src(path: &str, fns: &[Function]) {
```

Comment:
Wouldn't a macro be a better tool for the job of autogenerating code?

Reply:
Yes, I didn't use a macro for practical reasons.

1. I'm not good at them.
2. The errors that would've been produced would've likely cost me a lot of time to debug through a macro. I've found macro errors to be very mysterious.

Reply:
I haven't written Rust macros myself (yet), but I have maintained code like this that autogenerated code by printing strings and it was not fun.

Reply:
Ack. I think in terms of maintainability from myself, I prefer `Normal > Codegen > Macro`.

FWIW: I do this a lot at work (although with arguably better tooling) and its not ideal but not among the worst things we do.